### PR TITLE
Resolves #802 Warn when unknown entry types are used inside Bibtex/Biblatex

### LINKED ISSUE #802: Warn when unknown entry types are used inside Bibtex/Biblatex database
For example if one uses entry types of Biblatex inside a bibtex-defined database.

### PR DESCRIPTION
**Pull request by stefan-kolb**
Quick shot at the checker. WDYT?

### CODE REVIEW

**Comment by stefan-kolb** [comment]
I'm not sure if we 100% need this check just thinking...

File: src/main/java/net/sf/jabref/logic/integrity/BibTeXEntryTypeChecker.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * BibTeX mode only checker
+ */
+public class BibTeXEntryTypeChecker implements Checker {
+    private static final List<EntryType> BIBTEX = BibtexEntryTypes.ALL;
+    private static final List<EntryType> BIBLATEX = BibLatexEntryTypes.ALL;
+    private static final List<String> EXCLUSIVE_BIBLATEX = filterEntryTypesNames(BIBLATEX, isNotIncludedIn(BIBTEX));
+
+    public BibTeXEntryTypeChecker() {
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/integrity/BibTeXEntryTypeChecker.java]: Why empty constructor?

File: src/main/java/net/sf/jabref/logic/integrity/BibTeXEntryTypeChecker.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import net.sf.jabref.model.entry.BibtexEntryTypes;
+import net.sf.jabref.model.entry.EntryType;
+
+/**
+ * BibTeX mode only checker
+ */
+public class BibTeXEntryTypeChecker implements Checker {
+    private static final List<EntryType> BIBTEX = BibtexEntryTypes.ALL;
+    private static final List<EntryType> BIBLATEX = BibLatexEntryTypes.ALL;
+    private static final List<String> EXCLUSIVE_BIBLATEX = filterEntryTypesNames(BIBLATEX, isNotIncludedIn(BIBTEX));
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/integrity/BibTeXEntryTypeChecker.java]: I don't see the need for these static lists and would favor if the "exclusive biblatex" list is constructed in the check method.

File: src/main/java/net/sf/jabref/logic/integrity/BibTeXEntryTypeChecker.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * Will check if the current library uses any entry types from another mode.
+     * For example it will warn the user if he uses entry types defined for Biblatex inside a BibTeX library.
+     */
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        if (EXCLUSIVE_BIBLATEX.contains(entry.getType())) {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/integrity/BibTeXEntryTypeChecker.java]: I think `if (EntryTypes.getType(entry.getType(), Current BibDatabaseMode) == false)` also detects undefined entry types in BibLatexMode

File: src/main/java/net/sf/jabref/logic/integrity/BibTeXEntryTypeChecker.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            return Collections.singletonList(
+                    new IntegrityMessage(Localization.lang("Entry type %0 is only defined for Biblatex but not for BibTeX", entry.getType()), entry, "bibtexkey")
+            );
+        }
+        return Collections.emptyList();
+    }
+
+    private static List<String> filterEntryTypesNames(List<EntryType> types, Predicate<EntryType> predicate) {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/integrity/BibTeXEntryTypeChecker.java]: In my opinion, it makes sense to move this logic to the EntryTypes class and provide a `isExclusiveBibLatex(String type)` method there.

**Comment by lenhard** [comment]
Ok, looks good to me code-wise. Since this had been reviewed by two other devs and you fixed their issues, I will merge this right away.

**Pull request merged**